### PR TITLE
Give more icons to replace anons with

### LINKED ISSUE #263: Abstract-ify Satsukimous
>Would it be possible to make Satsukimous now abstract and allow us to use a custom image (either local or online) as the anon image?

### LINKED ISSUE #382: Option to change Anon avatars to anything we want.

### PR DESCRIPTION
**Pull request by Wolvan**
Add Matoi and Mako as icon replacement
Add Anonymous option to not replace the icon
Closes #382
Closes #263

### CODE REVIEW

File: Extensions/satsukimous.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //* TITLE Satsukimous **//
-//* VERSION 1.0.4 **//
-//* DESCRIPTION Turns anon icons into Satsuki **//
+//* VERSION 1.1.0 **//
+//* DESCRIPTION Turns anon icons into Satsuki, Matoi or Mako **//
```
Comment (bvtsang) [inline Extensions/satsukimous.js]: The names in the show are:

- Ryūko Matoi (纏 流子 Matoi Ryūko)
- Satsuki Kiryūin (鬼龍院 皐月 Kiryūin Satsuki)
- Mako Mankanshoku (満艦飾 マコ Mankanshoku Mako)

To keep with the naming convention, we should call Matoi either Ryuuko, Ryuko, or Ryūko.

File: Extensions/satsukimous.js [hunk 2]
```diff
@@ -11,8 +11,20 @@ XKit.extensions.satsukimous = new Object({
...
+			values: [
+				"Satsuki Kiryūin", "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+				"Matoi Ryūko", "http://38.media.tumblr.com/avatar_2e71003ae267_128.png",
+				"Mako Mankanshoku", "https://33.media.tumblr.com/avatar_759f9349bfc2_128.png",
+				"Anyonymous", "https://secure.assets.tumblr.com/images/anonymous_avatar_128.gif",
+			],
+		},
 		"play_scream": {
-			text: "matoi RYUKOOOOoO",
+			text: "matoi RYūKOOOOoO",
```
Comment (bvtsang) [inline Extensions/satsukimous.js]: Nitpick: breaking upper-case looks weird

**Comment by bvtsang** [comment]
If we're going to allow the user to pick from a set of replacement icons, we might as well abstractify this extension and allow the user to choose their own icon from anywhere on the Internet. This should solve #263.

File: Extensions/satsukimous.js [hunk 3]
```diff
@@ -11,8 +11,20 @@ XKit.extensions.satsukimous = new Object({
...
 
 	preferences: {
+		replacement: {
+			text: "Replacement Avatar",
+			default: "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+			value: "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+			type: "combo",
+			values: [
+				"Satsuki Kiryūin", "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+				"Matoi Ryūko", "http://38.media.tumblr.com/avatar_2e71003ae267_128.png",
```
Comment (bvtsang) [inline Extensions/satsukimous.js]: To keep with the naming convention, this should be Ryūko Matoi.

Comment (ThePsionic) [inline Extensions/satsukimous.js]: Somehow I cannot believe that we are talking about Kill La Kill characters
in the most serious way possible.

**Comment by Wolvan** [comment]
Custom replacement images are possible now, I also fixed the naming issues we had earlier.
Furthermore, the extension properly reverts it's changes on destroy

File: Extensions/satsukimous.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 //* TITLE Satsukimous **//
-//* VERSION 1.0.4 **//
-//* DESCRIPTION Turns anon icons into Satsuki **//
+//* VERSION 1.1.0 **//
+//* DESCRIPTION Turns anon icons into Satsuki, Ryuko or Mako or a custom image **//
```
Comment (dlmarquis) [inline Extensions/satsukimous.js]: It's a minor grammar nitpick, but this should be "Turns anon icons into Satsuki, Ryuko, Mako or a custom image"

Comment (Wolvan) [inline Extensions/satsukimous.js]: Right, thanks for pointing that out. Changed now

File: Extensions/satsukimous.js [hunk 5]
```diff
@@ -11,20 +11,51 @@ XKit.extensions.satsukimous = new Object({
...
+		replacement: {
+			text: "Replacement Avatar",
+			default: "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+			value: "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+			type: "combo",
+			values: [
+				"Satsuki Kiryūin", "https://31.media.tumblr.com/avatar_0bc380bccba7_128.png",
+				"Ryūko Matoi", "http://38.media.tumblr.com/avatar_2e71003ae267_128.png",
+				"Mako Mankanshoku", "https://33.media.tumblr.com/avatar_759f9349bfc2_128.png",
+				"Anyonymous", "https://secure.assets.tumblr.com/images/anonymous_avatar_128.gif",
```
Comment (bvtsang) [inline Extensions/satsukimous.js]: This should be spelled Anonymous.

**Comment by bvtsang** [comment]
Screenshot of Satsukimous's Custom Replacement option.

The wording "Custom Replacement" is ambiguous. Would it be better to call it "Custom Image URL"?

**Comment by Wolvan** [comment]
Thanks for the advice, @bvtsang, changed the file accordingly

**Comment by bvtsang** [comment]
LGTM

@homu r+

**Pull request merged**